extensions/parquet/table/src/main/java/io/deephaven/parquet/table/pagestore/topage/ToInstantPage.java

```diff
@@ -16,20 +22,49 @@
 
 public class ToInstantPage<ATTR extends Any> implements ToPage<ATTR, long[]> {
 
-    private static final ToInstantPage INSTANCE = new ToInstantPage<>();
+    @SuppressWarnings("unchecked")
+    public static <ATTR extends Any> ToPage<ATTR, Instant[]> create(
+            final Class<?> nativeType,
+            final LogicalTypeAnnotation.TimeUnit unit) {
+        if (nativeType == null || Instant.class.equals(nativeType)) {
+            switch (unit) {
+                case MILLIS:
+                    return FROM_MILLIS;
+                case MICROS:
+                    return FROM_MICROS;
+                case NANOS:
+                    return FROM_NANOS;
+                default:
+                    throw new IllegalArgumentException("Unsupported unit=" + unit);
+            }
```

Comment:
I think we're still keeping half the logic in the implementations and half in `ParquetColumnLocation`. I think we might want to make the *decisions* over in PCL.